app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -0,0 +1,429 @@
+    }
+
+    /**
+     * Setup the map to the desired look
+     */
+    fun setUpMap() {
+        //Restoring the map state
+        restoreCameraState()
+        restoreMapState()
+        setMapStyle()
+        //setBoundaries()
+        //setMinAndMaxZoom()
+
+        //To deactivate the 3d buildings
+        //map!!.isBuildingsEnabled = false
+    }
+
+    /**
+     * Restores the camera to the location it was before changing fragment or activity, goes to a initial position if it is the first time the map is opened
+     */
+    fun restoreCameraState() {
+        map!!.moveCamera(CameraUpdateFactory.newLatLngZoom(cameraPosition,cameraZoom))
+    }
+
+
+
+    /**
+     * Redraws all areas that were previously drawn before changing fragment or activity, draws some example
+     */
+    fun restoreMapState() {
+        if (areasPoints.isNotEmpty()) {
+            val temp = areasPoints.toMap()
+            areasPoints.clear()
+            //Draw all areas and points
+            for ((k, v) in temp) {
+                addArea(k, v.second.points, v.first.title)
+            }
+
+        } else {
+            //-----------Create example areas---------------
+            val listEvent1 = arrayListOf<LatLng>()
+            listEvent1.add(LatLng(46.52100506978624, 6.565499156713487))
+            listEvent1.add(LatLng(46.52073238207864, 6.565499156713487))
+            listEvent1.add(LatLng(46.52073238207864, 6.565711721777915))
+            listEvent1.add(LatLng(46.52100506978624, 6.565711721777915))
+            addArea("uid++.toString()", listEvent1, "Sushi Demo")
+
+            val listEvent2 = arrayListOf<LatLng>()
+            listEvent2.add(LatLng(46.52015447340308, 6.5656305849552155))
+            listEvent2.add(LatLng(46.52036049105315, 6.5658414736390105))
+            listEvent2.add(LatLng(46.52013394080612, 6.566103324294089))
+            addArea(uid++.toString(), listEvent2, "Triangle")
+
+            val listEvent3 = arrayListOf<LatLng>()
+            listEvent3.add(LatLng(46.52111073013754, 6.565624214708805))
+            listEvent3.add(LatLng(46.52107750943789, 6.565624214708805))
+            listEvent3.add(LatLng(46.52108443041863, 6.566078178584576))
+            listEvent3.add(LatLng(46.521115113422766, 6.5660761669278145))
+            listEvent3.add(LatLng(46.521115113422766, 6.565871313214302))
+            listEvent3.add(LatLng(46.52115986905187, 6.565871313214302))
+            listEvent3.add(LatLng(46.52115986905187, 6.565824374556541))
+            listEvent3.add(LatLng(46.521115113422766, 6.565824374556541))
+            addArea(uid++.toString(), listEvent3, "La route en T")
+        }
+    }
+
+    /**
+     * Changes the style of the map
+     */
+    fun setMapStyle() {
+        map!!.setMapStyle(MapStyleOptions(context!!.resources.getString(R.string.style_test3)))
+    }
+
+    /**
+     * Helper method to add a area to the map and generate an invisible marker in its center to display the area infos
+     */
```

Comment:
The arguments are self-explanatory, but keep the docs consistent with the rest it would be nice to add the "@param id : blablabla".